models/DataObject/Fieldcollection/Dao.php

```diff
@@ -96,7 +96,7 @@ public function load(DataObject\Concrete $object)
                             $collection->setValue($key, $value);
 
                             if ($collection instanceof DataObject\DirtyIndicatorInterface) {
-                                $collection->markFieldDirty($key, false);
+                                $collection->markFieldDirty($key, true);
```

Comment:
what is the exact reason for this? after the load operation the data shouldn't be dirty, right ?